### DISCUSSION TITLE
Type narrowing is lost when use selectFromResult

### DISCUSSION THREAD

**Discussion by arlgnv** [reduxjs/redux-toolkit#3015] [Q&A] [answered]
When I don't use `selectFromResult` I can, for example, write `if (isLoading)` and inside if block my `data` variable won't have undefined type. But when I use `selectFromResult` this behavior is lost. Maybe I am doing smth incorrectly?
![image](https://user-images.githubusercontent.com/46949445/208430808-9f8551b8-0f54-42e1-8628-ef8afb622a75.png)

**Comment by phryneas** [comment, marked as answer]
You are replacing the return type of the RTKQ query hook with your own (in this case implicit) type. 
If you want type behaviour like that, you will have to create a type with that behaviour and explicitly return that.

E.g.
```ts
type MyType = { isVaultBeingFetched: true, vault: undefined } | { isVaultBeingFetched: false, vault: SomeVaultType }
```
